libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3416,7 +3416,9 @@ void GCS_MAVLINK::handle_odometry(const mavlink_message_t &msg)
     const uint32_t timestamp_ms = correct_offboard_timestamp_usec_to_ms(m.time_usec, PAYLOAD_SIZE(chan, ODOMETRY));
     visual_odom->handle_vision_position_estimate(m.time_usec, timestamp_ms, m.x, m.y, m.z, q, posErr, angErr, m.reset_counter);
 
-    const Vector3f vel{m.vx, m.vy, m.vz};
+    // convert velocity vector from FRD to NED frame
+    Vector3f vel{m.vx, m.vy, m.vz};
+    vel.rotate_xy(q.get_euler_yaw());
```

Comment:
why only rotate in XY plane? why not a full FRD -> NED rotation?

Reply:
Indeed, it does seem to be a full rotation instead of just a yaw rotation.